reascripts/ReaSpeech/source/ASRControls.lua

```diff
@@ -0,0 +1,157 @@
+--[[
+
+ASRControls.lua - Controls/configuration for ASR plugin
+
+]]--
+
+ASRControls = Polo {
+  DEFAULT_TAB = 'asr-simple',
+
+  DEFAULT_LANGUAGE = '',
+  DEFAULT_MODEL_NAME = 'small',
+
+  SIMPLE_MODEL_SIZES = {
+    {'Small', 'small'},
+    {'Medium', 'medium'},
+    {'Large', 'distil-large-v3'},
+  },
+
+  new = function(plugin)
+    return {
+      plugin = plugin
+    }
+  end,
+}
+
+function ASRControls:init()
+  assert(self.plugin, 'ASRControls: plugin is required')
+
+  Logging.init(self, 'ASRControls')
+
+  self.language = ReaSpeechCombo.new {
+    default = self.DEFAULT_LANGUAGE,
+    label = 'Language',
+    items = WhisperLanguages.LANGUAGE_CODES,
+    item_labels = WhisperLanguages.LANGUAGES
+  }
+
+  self.translate = ReaSpeechCheckbox.new {
+    default = false,
+    label_long = 'Translate to English',
+    label_short = 'Translate',
+    width_threshold = ReaSpeechControlsUI.NARROW_COLUMN_WIDTH
+  }
+
+  self.hotwords = ReaSpeechTextInput.simple('', 'Hot Words')
+  self.initial_prompt = ReaSpeechTextInput.simple('', 'Initial Prompt')
+  self.model_name = ReaSpeechTextInput.simple(self.DEFAULT_MODEL_NAME, 'Model Name')
+
+  self.vad_filter = ReaSpeechCheckbox.new {
+    default = true,
+    label_long = 'Voice Activity Detection',
+    label_short = 'VAD',
+    width_threshold = ReaSpeechControlsUI.NARROW_COLUMN_WIDTH
+  }
+
+  self.model_name_buttons = ReaSpeechButtonBar.new {
+    default = self.DEFAULT_MODEL_NAME,
+    label = 'Model Name',
+    buttons = self.SIMPLE_MODEL_SIZES,
+    column_padding = ReaSpeechControlsUI.COLUMN_PADDING,
+    margin_bottom = ReaSpeechControlsUI.MARGIN_BOTTOM,
+    margin_left = ReaSpeechControlsUI.MARGIN_LEFT,
+    margin_right = ReaSpeechControlsUI.MARGIN_RIGHT,
+  }
+  self.model_name_buttons.on_set = function()
+    self.model_name:set(self.model_name_buttons:value())
+  end
+
+  self:init_layouts()
+end
+
+function ASRControls:tabs()
+  return {
+    ReaSpeechPlugins.tab('asr-simple', 'ASR - Simple',
+      function() self:render_simple() end),
+    ReaSpeechPlugins.tab('asr-advanced', 'ASR - Advanced',
+      function() self:render_advanced() end),
+  }
+end
```

Comment:
this is the meat of what `ReaSpeechPlugins` is looking for with "tabs" - a "tab" here being a key, label & rendering callback.